solution/q1/README.md

```diff
@@ -0,0 +1,224 @@
+# Solution to question 1
+
+## Terraform
+
+```
+cd terraform
+```
+
+### v1.5.0+
+
+New [config-driven import](https://www.hashicorp.com/blog/terraform-1-5-brings-config-driven-import-and-checks) allows to declare what resources are planned for import via `import` blocks.
+Their code is generated at plan step with the help of `-generate-config-out` argument.
+Once plan is applied, new (generated) code should be committed to the repository.
+
+Configuration for both buckets is split between 2 files: `main.tf` and `annalise_ai_generated.tf`.
+I am using config-driven import to acquire Annalise AI bucket into TF state. The workflow assumes 
+that only `main.tf` exists initially, later once plan with new generated code is applied (and annalise resources are confirmed to be present in the state via `terraform state list`) - `annalise_ai_generated.tf`
+is committed into the repo.
+```
+terraform init
+terraform plan -generate-config-out=annalise_ai_generated.tf
+# review contents of annalise_ai_generated.tf
+terraform apply
+# confirm that bucket is present in TF state
+# commit annalise_ai_generated.tf into the repo
+```
+
+(To avoid naming conflicts, I used `cd-test` as bucket name prefix.)
+
+This setup assumes that both buckets are owned by same account, which is unlikely the real case.
```

Comment:
Hi @cdgz, one other question from our review meeting. It is indeed the case that the buckets are owned by different AWS accounts. In addition to this change in the import process, would anything need to change about how the file copy is executed, in order to accommodate the source and destination buckets being in a different account?

Reply:
I don't think the copy part would need any changes, as long as AWS credentials that are used for this script are linked to an IAM role that is granted with write access to destination bucket (and bucket policy allows that). Object ACLs are set to private by default, but can be overridden in-flight:
```
However, the access control list (ACL) is not preserved and is set to private for the user making the request. To override the default ACL setting, specify a new ACL when generating a copy request.
```
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3#Client.CopyObject